code/modules/power/tesla/energy_ball.dm

```diff
@@ -169,6 +169,8 @@
 	C.dust()
 
 /proc/tesla_zap(atom/source, zap_range = 3, power, zap_flags = ZAP_DEFAULT_FLAGS, list/shocked_targets)
+	if(source == null)
```

Comment:
You could do QDELETED instead since it checks if it exists or if it's a null

Reply:
Sure, why not